core/src/main/java/bisq/core/trade/protocol/bsq_swap/tasks/seller/ProcessTxInputsMessage.java

```diff
@@ -45,41 +47,53 @@ public ProcessTxInputsMessage(TaskRunner<BsqSwapTrade> taskHandler, BsqSwapTrade
     @Override
     protected void run() {
         try {
-            runInterceptHook();
+            TxInputsMessage message = checkNotNull((TxInputsMessage) protocolModel.getTradeMessage());
+            checkNotNull(message);
 
-            TxInputsMessage data = checkNotNull((TxInputsMessage) protocolModel.getTradeMessage());
-            checkNotNull(data);
+            List<RawTransactionInput> inputs = message.getBsqInputs();
+            checkArgument(!inputs.isEmpty(), "Buyers BSQ inputs must not be empty");
 
-            List<RawTransactionInput> inputs = data.getBsqInputs();
-            checkArgument(!inputs.isEmpty(), "inputs must not be empty");
             long sumInputs = inputs.stream().mapToLong(rawTransactionInput -> rawTransactionInput.value).sum();
-            // If taker is buyer the inputs are BSQ, otherwise BTC
-            checkArgument(sumInputs >= BsqSwapCalculation.getBuyersRequiredBsqInputs(trade).getValue(),
-                    "Buyers BSQ inputs are not sufficient");
-
-            long change = data.getBsqChange();
+            long buyersBsqInputAmount = BsqSwapCalculation.getBuyersBsqInputValue(trade, getBuyersTradeFee()).getValue();
+            checkArgument(sumInputs >= buyersBsqInputAmount,
+                    "Buyers BSQ input amount do not match our calculated required BSQ input amount");
+
+            DaoFacade daoFacade = protocolModel.getDaoFacade();
+            long numValidBsqInputs = inputs.stream()
+                    .map(input -> new TxOutputKey(txIdOfRawInputParentTx(input), (int) input.index))
+                    .filter(daoFacade::isTxOutputSpendable)
+                    .count();
+            checkArgument(inputs.size() == numValidBsqInputs,
+                    "Some of the buyers BSQ inputs are not from spendable BSQ utxos according to our DAO state data.");
+
+            long change = message.getBsqChange();
             checkArgument(change == 0 || Restrictions.isAboveDust(Coin.valueOf(change)),
                     "BSQ change must be 0 or above dust");
 
-            String buyersBtcAPayoutAddress = data.getBuyersBtcPayoutAddress();
-            checkNotNull(buyersBtcAPayoutAddress, "buyersBtcAPayoutAddress must not be null");
-            checkArgument(!buyersBtcAPayoutAddress.isEmpty(), "buyersBtcAPayoutAddress must not be empty");
+            String buyersBtcPayoutAddress = message.getBuyersBtcPayoutAddress();
+            checkNotNull(buyersBtcPayoutAddress, "buyersBtcPayoutAddress must not be null");
+            checkArgument(!buyersBtcPayoutAddress.isEmpty(), "buyersBtcPayoutAddress must not be empty");
```

Comment:
Would be good to check that the address is a valid address

Reply:
Hm... if the peer send an invalid address, the tx would fail. If its just a wrong address its the problem of the peer. Not sure if it really adds any security here.

Reply:
It's more that it would be in line with the rest of the verification. Better to fail early than failing on signing the tx at a later step.

Reply:
Added it